concepts/infrastructure-reconciliation/README.md

```diff
@@ -1,41 +1,196 @@
+- Customer Cluster CR
+- SubscriptionRequest CR
+
+### Cluster Managers and the Cluster CRs
+
+The Cluster Managers' main responsibility is to manage the respective Cluster CRs, this is tied to both their provisioning and providing an access to the created K8S clusters.
+
+First of all, an overarching Cluster CR will provide a common API for operations connected to support or infrastructure maintenance. It contains reference to the resource corresponding to the sub-type of cluster:
+```
+apiVersion: operator.kyma-project.io/v1alpha1
+kind: Cluster
+metadata:
+	name: managed-cluster
+	namespace: default
+spec:
+	gardenerClusterRef:
+		apiVersion: operator.kyma-project.io/v1alpha1
+		kind: GardenerCluster
+		name: managed-cluster
+		namespace: default
+*** or ***
+    customerClusterRef:
+		apiVersion: operator.kyma-project.io/v1alpha1
+		kind: CustomerCluster
+		name: customer-cluster
+		namespace: default	
+```
+
+Then, there will be two resources, first for the Gardener-based, managed clusters:
+```
+apiVersion: operator.kyma-project.io/v1alpha1
+kind: GardenerCluster
+metadata:
+	name: managed-cluster
+	namespace: default
+spec:
+	core:
+		purpose: "Production"
+		region: "eu-central-1"
+	extensions:
+		shootNetworkingFilterDisabled: true
+	oidcConfig:
+		clientID: "asdf"
+		groupsClaim: "asdf"
+		issuerURL: "asdf"
+		signingAlgs: ["adsf"]
+		usernameClaim: "asdf"
+		usernamePrefix: "asdf"
+	infrastructure:
+		machineType: "m5.2xlarge"
+		autoScalerMin: 5
+		autoScalerMax: 5
+		provider: aws
+		numberOfZones: 3
+	subscriptionSelector: "globalaccount:abcd-0001" || "shared"
+	kubeconfigs:
+	- secretRef: 
+		name: managed-cluster-admin-kubeconfig
+		namespace: default
+	  clusterRole:
+		rules:
+		- apiGroups: ["*"]
+		  resources: ["*"]
+		  verbs: ["*"]
+	administrators: ["user.id@sap.com", "other.id@sap.com"]
+```
+
+This contains a minimal set of values that are required to provision a new Gardener cluster. The resource also contains the `administrators` and `subscriptionSelector` fields that are required for the Binding creation and requesting a hyperscaler subscription respectively.
+
+And the second one for BYOC model:
+```
+customer-cluster
+
+apiVersion: operator.kyma-project.io/v1alpha1
+kind: CustomerCluster
+metadata:
+	name: customer-cluster
+	namespace: default
+spec:
+	adminKubeconfig: LS0tV1...
+	kubeconfigs:
+	- secretRef: customer-cluster-admin-kubeconfig
+	  clusterRole:
+		rules:
+		- apiGroups: ["*"]
+		  resources: ["*"]
+		  verbs: ["*"]
+```
+
+This is the simpler one, as there is no cluster provisioning happening in this flow most of the parameters can be dropped.
+
+### Issuing kubeconfigs
+
+As stated before, the separation of kubeconfigs and limiting access is needed to have a healthy, secure environment. Cluster Managers will provide a possibility to create RBAC-based kubeconfigs in both BYOC and Gardener clusters. This will provide operators with the required access to perform any necessary actions.
```

Comment:
I think a major flaw that the entire environment has is that operators managing other operator deliveries need almost administrative access since to avoid privilege escalations, they must hold all permissions they are trying to grant to an operator they deploy. This means that even though we can create RBAC-based kubeconfigs, module manager most likely will request (almost) full administrative access. Maybe something to discuss separately, but worth mentioning. @adityabhatia FYI this is what we observed in our POCs around integrating module manager in real life operators with tons of crds and RBACs

Reply:
Yeah, that was my observation as well. On the other hand - there won't be that many operators that require those privileges. In addition, all of the affected ones will be located in the Control Plane cluster (Lifecycle Manager, Compass, ...?), and the rest will run in the SKR cluster. I think we can reevaluate this matter with the security guys as quite a few things changed since the last discussion.